.evergreen/config_generator/generators/task_groups.py

```diff
@@ -0,0 +1,17 @@
+from shrub.v3.evg_project import EvgProject
+
+from config_generator.etc import utils
+
+
+def generate():
+    task_groups = []
+
+    for component in utils.all_components():
+        if hasattr(component, 'task_groups'):
+            component_name = utils.component_name(component)
+            print(f' - {component_name}')
+            task_groups += component.task_groups()
+
+    task_groups.sort(key=lambda v: v.name)
+    yaml = utils.to_yaml(EvgProject(task_groups=task_groups))
```

Comment:
A JSON string is a valid equivalent YAML document. You can use `json.dumps` and write that into a YAML file and it should just work. (Use of `indent=n` would be recommended, though)

Reply:
I confirmed Evergreen accepts a JSON equivalent of the emitted config files. However, due to parens and braces being placed on their own line (using the default formatter), I observe the number of lines in the generated files explodes by up to 400% or more compared their YAML equivalents when tested on a seperate branch containing more relocated functions/tasks. On top of being more human-readable (no paren/brace noise), I think it is more valuable to be consistent with (all?) other Evergreen config files and documentation that are currently use YAML.

CC @kevinAlbs Thoughts regarding use of JSON for Evergreen config files?

Reply:
I prefer YAML. The consistency with other drivers may be helpful to compare configurations with other drivers (e.g. some common configuration is in https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/config.yml).

It may be helpful to be able to manually edit the config. I often edit the config to run quick experiments without editing the Python scripts.